### PR TITLE
Aditya | Week5 - Implementation of Redux Assignment with Redux Toolkit

### PR DESCRIPTION
**Pull request by aditya-s-19**
Pull Request Questionnaire

(1) Briefly describe your change.
Have completed the counter with redux-toolkit assignment

(2) What parts of the application are affected by your change?
It'd just affect the Aditya Week 5 component

(3) How did you test your change? If relevant, which role(s) and/or customer configuration(s) did you use?
Did the basic checks and tests

### CODE REVIEW

File: src/components/Home/Aditya/index.tsx [hunk 1]
```diff
@@ -1,25 +1,35 @@
...
+          <button className={`weekButton ${selectedWeek === 'week4' ? 'active' : ''}`} value='week4' onClick={handleWeekClick}>
+            Week-4
+          </button>
+        </Link>
+        <Link to='week5'>
+          <button className={`weekButton ${selectedWeek === 'week5' ? 'active' : ''}`} value='week5' onClick={handleWeekClick}>
+            Week-5
+          </button>
+        </Link>
       </div>
```
Comment (RitikaGarg-ITT) [inline src/components/Home/Aditya/index.tsx:30]: instead writing redundant code you can use map to implement this

File: src/components/Home/Aditya/Week5/reducers/changeCounter.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const changeCounter = (state = initialValue, action: Action): number => {
+  switch (action.type) {
+    case 'INCREMENT':
+      return state + 1;
+    case 'DECREMENT':
+      return state - 1;
+    default:
+      return state;
+  }
+};
```
Comment (RitikaGarg-ITT) [inline src/components/Home/Aditya/Week5/reducers/changeCounter.ts]: It was suggested to use Redux Toolkit. You can also try for that.

File: src/components/Home/Aditya/Week5/components/counter/counterStyles.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import styled from 'styled-components';
+
+export const CounterWrapper = styled.div`
+  .App {
```
Comment (RitikaGarg-ITT) [inline src/components/Home/Aditya/Week5/components/counter/counterStyles.ts]: make a small

File: src/components/Home/Aditya/Week5/components/counter/Counter.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { CounterWrapper } from './counterStyles';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../../store';
+import { incrementCount, decrementCount } from '../../features/counterSlice';
+
+const Counter: React.FC = () => {
+  // eslint-disable-next-line
+  const count = useSelector((state: RootState) => state.counter);
```
Comment (anusreesubash) [inline src/components/Home/Aditya/Week5/components/counter/Counter.tsx:8]: why is lint disabled?

**Review by anusreesubash — DISMISSED** [review]

File: src/components/Home/Aditya/Week4/pages/Week3/components/searchUser/SearchUser.tsx [hunk 5]
```diff
@@ -40,3 +40,7 @@ const SearchUser: React.FC = () => {
 };
 
 export default SearchUser;
+
+//empty users
+//results overflow
```
Comment (Harrrrry) [inline src/components/Home/Aditya/Week4/pages/Week3/components/searchUser/SearchUser.tsx]: Remove commented code

File: src/components/Home/Aditya/Week4/pages/Week1/app.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import { setTodoShowTasks, setTodoCurrentFilter } from './features/todoSlice';
+
+const App: React.FC = () => {
+  const todo = useSelector((state: RootState) => state.todo);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    switch (todo.currentFilter) {
+      case 'Due Closest':
+      case 'Due Farthest': {
```
Comment (Harrrrry) [inline src/components/Home/Aditya/Week4/pages/Week1/app.tsx]: Here, I suggest to put the string in constants and use here

File: src/components/Home/Aditya/Week4/pages/Week3/utils/functions.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  let currentTask = str.replace(/\s+/g, ' ');
+  if (currentTask[0] === ' ') {
+    currentTask = currentTask.slice(1);
+  }
+  return currentTask;
+};
+
+export const getTodayDate = (): string => {
+  return new Date().toISOString().slice(0, 10);
+};
```
Comment (shreegovinda) [inline src/components/Home/Aditya/Week4/pages/Week3/utils/functions.ts]: whats the difference between this file and src/components/Home/Aditya/Week4/pages/Week1/utils/utils.ts

**Review by shreegovinda — DISMISSED** [review]

**Review by RitikaGarg-ITT — APPROVED** [review]

**Review by choudhary-keshav — APPROVED** [review]

**Pull request merged**